src/Microsoft.AspNet.Mvc.Core/ParameterBinding/ModelBindingHelper.cs

```diff
@@ -532,5 +532,17 @@ public static object ConvertValuesToCollectionType<T>(Type modelType, IList<T> v
                 return null;
             }
         }
+
+        private static Type[] GetTypeArgumentsIfMatch([NotNull] Type closedType, Type matchingOpenType)
```

Comment:
This and `GetPossibleBinderInstanceType` should just move to `GenericModelBinder` which is where they are used, and probably just simplified to do what they are supposed to do to support `KeyValuePair<>`. That's the only call I can find to either of these.